wcfsetup/install/files/lib/system/service/worker/Util.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace wcf\system\service\worker;
+
+use Jose\Component\Core\JWK;
+use ParagonIE\ConstantTime\Base64UrlSafe;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class Util
+{
+    /**
+     * Unserialize the public key into its x and y components.
+     *
+     * @param string $data
+     * @return array{x: string, y: string}
+     */
+    public static function unserializePublicKey(string $data): array
+    {
+        if (\mb_strlen($data, '8bit') !== VAPID::PUBLIC_KEY_LENGTH || $data[0] !== "\x04") {
+            throw new \InvalidArgumentException('Invalid public key format.');
+        }
+        $data = \mb_substr($data, 1, null, '8bit');
+        $dataLength = \mb_strlen($data, '8bit');
+
+        [$x, $y] = \mb_str_split($data, \intdiv($dataLength, 2), '8bit');
+        return [
+            'x' => $x,
+            'y' => $y,
+        ];
+    }
+
+    /**
+     * Serialize the public key.
+     *
+     * @param string $x encoded base64 x coordinate
+     * @param string $y encoded base64 y coordinate
+     * @return string
+     */
+    public static function serializePublicKey(string $x, string $y): string
+    {
+        $x = Base64UrlSafe::decodeNoPadding($x);
+        $y = Base64UrlSafe::decodeNoPadding($y);
+        $result = "\x04";
+        $result .= \str_pad($x, 32, "\x00", STR_PAD_LEFT);
+        return $result . \str_pad($y, 32, "\x00", STR_PAD_LEFT);
+    }
+
+    /**
+     * Creates a JWK with EC P-256 curve.
+     *
+     * @param string|null $x non-encode base64 x coordinate
+     * @param string|null $y non-encoded base64 y coordinate
+     * @param string|null $d base64 encoded private key
+     *
+     * @return JWK
+     */
+    public static function createJWK(?string $x = null, string $y = null, #[\SensitiveParameter] ?string $d = null): JWK
```

Comment:
see: https://wiki.php.net/rfc/deprecate-implicitly-nullable-types
```suggestion
    public static function createJWK(?string $x = null, ?string $y = null, #[\SensitiveParameter] ?string $d = null): JWK
```

I wonder though: Are the `= null` default values even meaningful? Having an `x` without a `y` does not really make sense to me.